From 9be7b53b225aa8b6a9ed71cfa56bedf8b465075c Mon Sep 17 00:00:00 2001
From: Ramesh Babu <ramesh.babu@intel.com>
Date: Fri, 9 Oct 2015 01:42:16 +0530
Subject: [PATCH 1774/2508] ASoC: Intel: update broxton-p machine driver

commit d8630bc48f1d2a18502879c74c0938e5de7fc33b from
https://github.com/01org/linux-apollolake-i

Update the Broxton-P machine driver's map
and dai-link paramets as per the board.

Change-Id: Ib857ca1762c2de4463ed02621b42d06bcfeee412
Signed-off-by: Ramesh Babu <ramesh.babu@intel.com>
Reviewed-on: https://git-gar-1.devtools.intel.com/gerrit/7514
---
 sound/soc/intel/boards/bxt_rt298.c |   61 +++++++++--------------------------
 1 files changed, 16 insertions(+), 45 deletions(-)

diff --git a/sound/soc/intel/boards/bxt_rt298.c b/sound/soc/intel/boards/bxt_rt298.c
index b2b63b9..157301a 100644
--- a/sound/soc/intel/boards/bxt_rt298.c
+++ b/sound/soc/intel/boards/bxt_rt298.c
@@ -1,12 +1,10 @@
 /*
- * Intel Broxton I2S Machine Driver
+ * Intel Broxton-P I2S Machine Driver
  *
  * Copyright (C) 2014-2015, Intel Corporation. All rights reserved.
  *
  * Modified from:
- *   Intel Broadwell Wildcatpoint SST Audio
- *
- *   Copyright (C) 2013, Intel Corporation. All rights reserved.
+ *   Intel Skylake I2S Machine driver
  *
  * This program is free software; you can redistribute it and/or
  * modify it under the terms of the GNU General Public License version
@@ -60,7 +58,7 @@ static const struct snd_soc_dapm_route broxton_rt298_map[] = {
 	{"Speaker", NULL, "SPOR"},
 	{"Speaker", NULL, "SPOL"},
 
-	/* HP jack connectors - unknown if we have jack deteck */
+	/* HP jack connectors - unknown if we have jack detect */
 	{"Headphone Jack", NULL, "HPO Pin"},
 
 	/* other jacks */
@@ -71,21 +69,17 @@ static const struct snd_soc_dapm_route broxton_rt298_map[] = {
 	{"DMIC AIF", NULL, "SoC DMIC"},
 
 	/* CODEC BE connections */
-	{ "AIF1 Playback", NULL, "ssp0 Tx"},
-	{ "ssp0 Tx", NULL, "codec0_out"},
-	{ "ssp0 Tx", NULL, "codec1_out"},
+	{ "AIF1 Playback", NULL, "ssp5 Tx"},
+	{ "ssp5 Tx", NULL, "codec0_out"},
 
-	{ "codec0_in", NULL, "ssp0 Rx" },
-	{ "codec1_in", NULL, "ssp0 Rx" },
-	{ "ssp0 Rx", NULL, "AIF1 Capture" },
+	{ "codec0_in", NULL, "ssp5 Rx" },
+	{ "ssp5 Rx", NULL, "AIF1 Capture" },
 
 	{ "dmic01_hifi", NULL, "DMIC01 Rx" },
 	{ "dmic23_hifi", NULL, "DMIC23 Rx" },
 	{ "DMIC01 Rx", NULL, "Capture" },
 	{ "DMIC23 Rx", NULL, "Capture" },
 
-	{ "hif1", NULL, "iDisp Tx"},
-	{ "iDisp Tx", NULL, "iDisp_out"},
 
 };
 
@@ -107,7 +101,7 @@ static int broxton_rt298_codec_init(struct snd_soc_pcm_runtime *rtd)
 }
 
 
-static int broxton_ssp0_fixup(struct snd_soc_pcm_runtime *rtd,
+static int broxton_ssp5_fixup(struct snd_soc_pcm_runtime *rtd,
 			struct snd_pcm_hw_params *params)
 {
 	struct snd_interval *rate = hw_param_interval(params,
@@ -119,7 +113,7 @@ static int broxton_ssp0_fixup(struct snd_soc_pcm_runtime *rtd,
 	rate->min = rate->max = 48000;
 	channels->min = channels->max = 2;
 
-	/* set SSP0 to 16 bit */
+	/* set SSP5 to 16 bit */
 	snd_mask_set(&params->masks[SNDRV_PCM_HW_PARAM_FORMAT -
 				    SNDRV_PCM_HW_PARAM_FIRST_MASK],
 				    SNDRV_PCM_FORMAT_S16_LE);
@@ -133,7 +127,7 @@ static int broxton_rt298_hw_params(struct snd_pcm_substream *substream,
 	struct snd_soc_dai *codec_dai = rtd->codec_dai;
 	int ret;
 
-	ret = snd_soc_dai_set_sysclk(codec_dai, RT298_SCLK_S_PLL, 24000000,
+	ret = snd_soc_dai_set_sysclk(codec_dai, RT298_SCLK_S_PLL, 19200000,
 		SND_SOC_CLOCK_IN);
 	if (ret < 0) {
 		dev_err(rtd->dev, "can't set codec sysclk configuration\n");
@@ -187,26 +181,13 @@ static struct snd_soc_dai_link broxton_rt298_dais[] = {
 		.nonatomic = 1,
 		.dynamic = 1,
 	},
-	/*{
-		.name = "Skl HDMI Port",
-		.stream_name = "Hdmi",
-		.cpu_dai_name = "System Pin",
-		.codec_name = "snd-soc-dummy",
-		.codec_dai_name = "snd-soc-dummy-dai",
-		.platform_name = "0000:00:0e.0",
-		.dpcm_playback = 1,
-		.init = NULL,
-		.ignore_suspend = 1,
-		.nonatomic = 1,
-		.dynamic = 1,
-	},*/
 
 	/* Back End DAI links */
 	{
-		/* SSP0 - Codec */
-		.name = "SSP0-Codec",
+		/* SSP5 - Codec */
+		.name = "SSP5-Codec",
 		.be_id = 0,
-		.cpu_dai_name = "SSP0 Pin",
+		.cpu_dai_name = "SSP5 Pin",
 		.platform_name = "0000:00:0e.0",
 		.no_pcm = 1,
 		.codec_name = "i2c-INT343A:00",
@@ -216,7 +197,7 @@ static struct snd_soc_dai_link broxton_rt298_dais[] = {
 			SND_SOC_DAIFMT_CBS_CFS,
 		.ignore_suspend = 1,
 		.ignore_pmdown_time = 1,
-		.be_hw_params_fixup = broxton_ssp0_fixup,
+		.be_hw_params_fixup = broxton_ssp5_fixup,
 		.ops = &broxton_rt298_ops,
 		.dpcm_playback = 1,
 		.dpcm_capture = 1,
@@ -243,17 +224,6 @@ static struct snd_soc_dai_link broxton_rt298_dais[] = {
 		.dpcm_capture = 1,
 		.no_pcm = 1,
 	},
-	/*{
-		.name = "iDisp",
-		.be_id = 3,
-		.cpu_dai_name = "iDisp Pin",
-		.codec_name = "codec#002.2",
-		.codec_dai_name = "intel-hdmi-hif1",
-		.platform_name = "0000:00:0e.0",
-		.dpcm_playback = 1,
-		.ignore_suspend = 1,
-		.no_pcm = 1,
-	},*/
 };
 
 /* broxton audio machine driver for SPT + RT298S */
@@ -295,7 +265,8 @@ static struct platform_driver broxton_audio = {
 module_platform_driver(broxton_audio)
 
 /* Module information */
-MODULE_AUTHOR("Omair Mohammed Abdullah <omair.m.abdullah@intel.com>");
+MODULE_AUTHOR("Ramesh Babu <Ramesh.Babu@intel.com>");
+MODULE_AUTHOR("Senthilnathan Veppur <senthilnathanx.veppur@intel.com>");
 MODULE_DESCRIPTION("Intel SST Audio for Broxton");
 MODULE_LICENSE("GPL v2");
 MODULE_ALIAS("platform:bxt_alc298s_i2s");
-- 
1.7.5.4

